### DISCUSSION TITLE
Is it possible to trigger configuration (facetFilters) update on URL change?

### DISCUSSION THREAD

**Discussion by jsdmc** [algolia/instantsearch#5330] [Q&A]
Hello guys.
I'm having the following situation. 
I needed to implement AN + OR search behaviour. None of the widgets and hooks allowed me to accomplish that query for selected attribute. So I decided to go with next approach - updating `facetFilters` in configuration using `useConfigure` hook.

Something like that
```
facetFilters: [
  'options.id: 33',
  ['options.id:24', 'options.id:9'],
]
```
which corresponds to search query `options.id = 33 AND (options.id = 24 OR options.id = 9)`

To reflect that changes in URL I had to provide custom state mapping because default one `simpleStateMapping ` omits configuration changes.
https://github.com/algolia/instantsearch.js/blob/master/src/lib/stateMappings/simple.ts#L6

Initial search loading from URL with any combination of filters from configuration and other widgets works perfect. But next scenario doesn't work

1. In the app I set this filters `options.id = 33 AND options.id = 24`. Url reflect these changes. Search results correct.
2. Then update applied filters to `options.id = 33 AND (options.id = 24 OR options.id = 9)`. Url is correct. Search results correct.
3. Press back button in browser. Now url corresponds to `options.id = 33 AND options.id = 24` but `indexUiState.configure` is still contains state for `options.id = 33 AND (options.id = 24 OR options.id = 9)`. Search results incorrect.

Expected behaviour: after URL update `facetFilters` in configuration (and whole `configure` object) should be also updated.

Is it a bug or the limitation of architecture of widgets that can't be fixed?

Thanks for help

**Comment by felipepastor** [comment]
Not related, but thanks for describing that we can use `useConfigure` to search using `aroundLatLng` and many things more. I was looking into `geoSearch` 🤦 

I'm also updating my URL based on the object and what I did was to overwrite the `routing` and play with the `stateToMapping` and `routeToState` and when I click on the back button it reflects on my results too.

```
          routing={{
            router: history({
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore
              getLocation(): Location | URL {
                return typeof window === 'undefined'
                  ? new URL(url)
                  : window.location;
              },
            }),
            stateMapping: {
              stateToRoute(uiState) {
                const { geoSearch, configure, refinementList, ...rest } =
                  uiState[INDEX];

                ...
              },
              routeToState(routeState) {
                ...
              },
            },
          }}
```

**Reply by jsdmc** [reply]
In my case back button works as expected when going through some filters that are saved not in `configure` part of undexUiState. So it's not related to `getLocation `.
